### PR TITLE
feat(gallery): query to hide burned nft

### LINKED ISSUE #679: Do not show burned NFTs in the UI
**Is your feature request related to a problem? Please describe.**

It is completely confusing to display burned NFTs in the gallery views. Most platforms do not do this because once something is burned it exists on-chain only, and there is no good reason to keep displaying that token in the UI.

What's worse, is that if I create a few new tokens in experimentation which are then burned, these worthless tokens are cluttering up my profile and confusing both buyers and collectors.

**Describe the solution you'd like**

Hide burned tokens from all UI presentation views by default.

**Describe alternatives you've considered**

If people really want to see their burned tokens, a separate functionality could be added to provide for that.

**Additional context**

burned

The items with a 🔥 emoji on them in this screenshot should just not appear at all. This is pretty much the standard across all NFT platforms and so it would be nice to see it here too.

Thanks!

### PR DESCRIPTION
**Pull request by roiLeo**
I'm using for test : 
- /rmrk/u/FUEGFWoPPtX4XkRzEnZnMgwGKENscHXAiEZy6kcgaDz1BY6
- /rmrk/collection/800f8a914281765a7d-KITTY

### Before submitting this PR, please make sure:
- [x] Code builds clean without any errors or warnings
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I didn't break any original functionality

### Optional
- [x] I've tested it on mobile and everything works

### PR type
- [x] Feature

### What's new? (maybe part of changelog)
- closes #679 related
- [x] removed burned nft from Gallery
- [x] removed burned nft from User Collection
- [x] removed burned nft from Profile "Created"
- [x] removed burned nft from Profile "Collected"
- [x] removed burned nft from Profile "Sold"

Let me know if I've missed any UI that is showing burned nft.

### CODE REVIEW

**Comment by yangwao** [comment]
Don't forget to add url to profile where you are testing it. 
nft.kodadot.xyz one vs how it looks on PR to quick check :)

**Comment by vikiival** [comment]
Yup this could work! Good job

**Comment by yangwao** [comment]
@roiLeo others are there or are we waiting to be made? just curious! :)


 removed burned nft from User Collection
 removed burned nft from Profile "Collected"
 removed burned nft from Profile "Sold"

**Comment by roiLeo** [comment]
> 
> 
> @roiLeo others are there or are we waiting to be made? just curious! :)
> 
> removed burned nft from User Collection
> removed burned nft from Profile "Collected"
> removed burned nft from Profile "Sold"

Yup, this needs to be done. I will push some changes soon.

File: src/queries/collectionById.graphql [hunk 1]
```diff
@@ -3,12 +3,13 @@
 query collectionById($id: String!) {
   collectionEntity(id: $id) {
     ...collection
-    nfts {
+    nfts(filter: { burned: { distinctFrom: true } }) {
```
Comment (vikiival) [inline src/queries/collectionById.graphql:6]: Wow, smart move!

File: src/queries/collectionById.graphql [hunk 2]
```diff
@@ -3,12 +3,13 @@
...
       nodes {
         id
         metadata
         name
         price
+        burned
```
Comment (vikiival) [inline src/queries/collectionById.graphql:12]: maybe reuse `nft` fragment ? (`nft.graphql`)

**Review by vikiival — COMMENTED** [review]
Haven't tested but the implementation is really good here

File: src/queries/collectionById.graphql (hunk 2 above)
Comment (roiLeo) [reply src/queries/collectionById.graphql:12]: I'm not sure what you mean here by "reuse ```nft```fragment"

**Comment by yangwao** [comment]
paid out using our transfer by @prachi00 ;)
image

**Pull request merged**